vault/test_cluster.go

```diff
@@ -0,0 +1,5 @@
+//go:build !deadlock
```

Comment:
Mind renaming this file to something less generic, e.g. test_cluster_do_not_detect_deadlock.go?